test/index.test.ts

```diff
@@ -1,48 +1,110 @@
   const starkKeyPair = ec.getKeyPair(privKey);
   const account = new Account(defaultProvider, address, starkKeyPair);
 
-  it('StarkNetTx.propose()', async () => {
-    const author = walletAddress;
-    const metadataUri = 'ipfs://QmNrm6xKuib1THtWkiN5CKtBEerQCDpUtmgDqiaU2xDmca';
+  describe('vanilla authenticator', () => {
+    const client = new StarkNetTx();
+    const space = '0x03ddbcdfec756eb9e9e0fbce7a90721570a09fc97d31a15b8e40769b93ecb957';
+    const authenticator = '0x6ad07205a4d725c5c2b10c4f5fbdfaaa351c742fce7a5a22b2b56fd8d5afd62';
 
-    const receipt = await client.propose(account, author, space, metadataUri);
-    console.log('Receipt', receipt);
+    it('StarkNetTx.propose()', async () => {
+      const envelope = {
+        address: walletAddress,
+        sig: null,
+        data: {
+          message: {
+            space,
+            authenticator,
+            metadataURI: 'ipfs://QmNrm6xKuib1THtWkiN5CKtBEerQCDpUtmgDqiaU2xDmca',
+            executionParams: []
+          }
+        }
+      };
 
-    // await defaultProvider.waitForTransaction(receipt.transaction_hash);
+      const receipt = await client.propose(account, envelope);
+      console.log('Receipt', receipt);
 
-    expect(receipt.code).toBe('TRANSACTION_RECEIVED');
-  }, 300e3);
+      // await defaultProvider.waitForTransaction(receipt.transaction_hash);
 
-  it('StarkNetTx.vote()', async () => {
-    const voter = walletAddress;
-    const proposal = '1';
-    const choice = '1';
+      expect(receipt.code).toBe('TRANSACTION_RECEIVED');
+    }, 300e3);
 
-    const receipt = await client.vote(account, voter, space, proposal, choice);
-    console.log('Receipt', receipt);
+    it.only('StarkNetTx.vote()', async () => {
+      const envelope = {
+        address: walletAddress,
+        sig: null,
+        data: {
+          message: {
+            space,
+            authenticator,
+            proposal: 1,
+            choice: 1
+          }
+        }
+      };
 
-    // await defaultProvider.waitForTransaction(receipt.transaction_hash);
+      const receipt = await client.vote(account, envelope);
+      console.log('Receipt', receipt);
 
-    expect(receipt.code).toBe('TRANSACTION_RECEIVED');
-  }, 300e3);
+      // await defaultProvider.waitForTransaction(receipt.transaction_hash);
+
+      expect(receipt.code).toBe('TRANSACTION_RECEIVED');
+    }, 300e3);
+  });
+
+  describe('ethSig authenticator', () => {
+    const client = new StarkNetTx();
+    const ethSigClient = new EthereumSig(walletAddress);
+    const space = '0x03a48d8a7e80b7b295ef15efc4e1604e78b1f4460266cd460b6eb8826846bba0';
+    const authenticator = '0x594a81b66c3aa2c64577916f727e1307b60c9d6afa80b6f5ca3e3049c40f643';
+
+    it('StarkNetTx.propose()', async () => {
+      const envelope = await ethSigClient.propose(wallet, walletAddress, {
+        space,
+        authenticator,
+        metadataURI: 'ipfs://QmNrm6xKuib1THtWkiN5CKtBEerQCDpUtmgDqiaU2xDmca',
+        executionParams: []
```

Comment:
`executionParams` doesn't need to be dynamic yet (could be hardcodes as so is `executor`), but when it's part of input it can be used to to calculate executionParamsHash in `ethSigClient` as well as passed as part of calldata without hardcoding it in two different places.